falcon/asgi/app.py

```diff
@@ -767,7 +820,67 @@ async def _call_lifespan_handlers(self, ver, scope, receive, send):
                 await send({'type': EventType.LIFESPAN_SHUTDOWN_COMPLETE})
                 return
 
+    async def _handle_websocket(self, ver, scope, receive, send):
+        first_event = await receive()
+        if first_event['type'] != EventType.WS_CONNECT:
+            # The handshake was abandoned or this is a message we don't
+            #   support, so bail out.
+            await send({
+                'type': EventType.WS_CLOSE,
+                'code': WSCloseCode.SERVER_ERROR
+            })
+            return
+
+        req = self._request_type(scope, receive, options=self.req_options)
+
+        web_socket = WebSocket(
+            ver,
+            scope,
+            receive,
+            send,
+            self.ws_options.media_handlers,
+            self.ws_options.max_receive_queue,
+        )
+
+        on_websocket = None
+        params = {}
+
+        request_mw, resource_mw = self._middleware_ws
+
+        try:
+            for process_request_ws in request_mw:
+                await process_request_ws(req)
+
+            on_websocket, params, resource, req.uri_template = self._get_responder(req)
+
+            # NOTE(kgriffs): If the request did not match any
+            # route, a default responder is returned and the
+            # resource is None. In that case, we skip the
+            # resource middleware methods. Resource will also be
+            # None when a middleware method already set
+            # resp.complete to True.
+            if resource:
+                for process_resource_ws in resource_mw:
+                    await process_resource_ws(req, resource, params)
+
+            try:
+                await on_websocket(req, web_socket, **params)
+            except HTTPNotFound:
+                falcon._logger.warning(
```

Comment:
Good catch. I have removed this try...except since it essentially duplicates ``_python_error_handler()``, and have switched to using ``debug()`` with `%` based formatting.